[6.0.0] '--rulesdir' option doesn't work if a config file has 'root:true'

**Tell us about your environment**

* **ESLint Version:** 6.0.0
* **Node Version:** 12.2.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**

Default.

**Please show your full configuration:**

```js
module.exports = {
    root: true,
    rules: {
        test: "error"
    }
};
```

```js
// internal-rules/test.js
const { Linter } = require("eslint")
const linter = new Linter()
const rules = linter.getRules()

module.exports = rules.get("no-console")
```

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
console.log("hello")
```

```bash
eslint test.js --rulesdir internal-rules
```

**What did you expect to happen?**

The `test` rule in the `internal-rules` directory warns the code.

**What actually happened? Please include the actual, raw output from ESLint.**

```
C:\Users\t-nagashima.AD\dev\sandbox\test.js
  1:1  error  Definition for rule 'test' was not found  test

✖ 1 problem (1 error, 0 warnings)
```

**Are you willing to submit a pull request to fix this bug?**

Yes.

--- a/lib/cli-engine/cascading-config-array-factory.js
+++ b/lib/cli-engine/cascading-config-array-factory.js
@@ -313,6 +313,7 @@ class CascadingConfigArrayFactory {
 
         if (configArray.length > 0 && configArray.isRoot()) {
             debug("Stop traversing because of 'root:true'.");
+            configArray.unshift(...baseConfigArray);
             return this._cacheConfig(directoryPath, configArray);
         }
 

--- a/tests/lib/cli-engine/_utils.js
+++ b/tests/lib/cli-engine/_utils.js
@@ -314,6 +314,7 @@ function defineConfigArrayFactoryWithInMemoryFileSystem({
     // Override the default cwd.
     return {
         fs,
+        stubs,
         RelativeModuleResolver,
         ConfigArrayFactory: cwd === process.cwd
             ? ConfigArrayFactory
@@ -336,9 +337,9 @@ function defineCascadingConfigArrayFactoryWithInMemoryFileSystem({
     cwd = process.cwd,
     files = {}
 } = {}) {
-    const { fs, RelativeModuleResolver, ConfigArrayFactory } =
+    const { fs, stubs, RelativeModuleResolver, ConfigArrayFactory } =
            defineConfigArrayFactoryWithInMemoryFileSystem({ cwd, files });
-    const loadRules = proxyquire(LoadRulesPath, { fs });
+    const loadRules = proxyquire(LoadRulesPath, stubs);
     const { CascadingConfigArrayFactory } =
         proxyquire(CascadingConfigArrayFactoryPath, {
             "./config-array-factory": { ConfigArrayFactory },

--- a/tests/lib/cli-engine/cli-engine.js
+++ b/tests/lib/cli-engine/cli-engine.js
@@ -3421,6 +3421,38 @@ describe("CLIEngine", () => {
                 assert.fail("Expected to throw an error");
             });
         });
+
+        describe("with '--rulesdir' option", () => {
+            it("should use the configured rules which are defined by '--rulesdir' option.", () => {
+                const rootPath = getFixturePath("cli-engine/with-rulesdir");
+                const StubbedCLIEngine = defineCLIEngineWithInMemoryFileSystem({
+                    cwd: () => rootPath,
+                    files: {
+                        "internal-rules/test.js": `
+                            module.exports = context => ({
+                                ExpressionStatement(node) {
+                                    context.report({ node, message: "ok" })
+                                }
+                            })
+                        `,
+                        ".eslintrc.json": JSON.stringify({
+                            root: true,
+                            rules: { test: "error" }
+                        }),
+                        "test.js": "console.log('hello')"
+                    }
+                }).CLIEngine;
+
+                engine = new StubbedCLIEngine({
+                    rulePaths: ["internal-rules"]
+                });
+                const report = engine.executeOnFiles(["test.js"]);
+
+                assert.strictEqual(report.results.length, 1);
+                assert.strictEqual(report.results[0].messages.length, 1);
+                assert.strictEqual(report.results[0].messages[0].message, "ok");
+            });
+        });
     });
 
     describe("getConfigForFile", () => {

